LifeSpace/Debug/LogViewer.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  LogShareView.swift
+//  LifeSpace
+//
+//  Created by Vishnu Ravi on 11/8/24.
+//
+
+import OSLog
+import Spezi
+import SwiftUI
+
+
+struct LogViewer: View {
+    @Environment(LogManager.self) var manager
+    
+    @State private var startDate: Date = Calendar.current.date(byAdding: .day, value: -1, to: Date()) ?? Date()
+    @State private var endDate = Date()
+    @State private var selectedLogType: LogType = .all
+    @State private var logs = ""
+    @State private var isLoading = false
+    
+    var body: some View {
+        VStack {
+            /// Date range selection
+            HStack {
+                DatePicker("FROM", selection: $startDate, displayedComponents: .date)
+                Spacer()
+                DatePicker("TO", selection: $endDate, displayedComponents: .date)
+            }
+            .padding()
+            
+            /// Log type selection
+            Picker("LOG_TYPE", selection: $selectedLogType) {
+                ForEach(LogType.allCases) { type in
+                    Text(type.rawValue).tag(type)
+                }
+            }
+            .pickerStyle(SegmentedPickerStyle())
+            .padding()
+            
+            ScrollView {
+                if isLoading {
+                    ProgressView("LOADING_LOGS")
+                        .padding()
+                } else {
+                    Text(logs)
```

Comment:
We could use a content unavailable view in case there are no logs to display.

In addition, a huge Text view might be a bit inefficient in rendering a complete set of logs; we might want to use a list and display a log entry in each row of the list. Each row item could then do its formatting (we should pass along the log entry and maybe not a constructed string) and could then render things like date, log level, and others in a nicely formatted form.

Reply:
Thanks! I updated this to be a `List` and am passing in an array of `OSLogEntryLog`s that are formatted and rendered there.